mythril/analysis/modules/state_change_external_calls.py

```diff
@@ -0,0 +1,197 @@
+
+    def __init__(self, call: Call, user_defined_address: bool) -> None:
+        self.call = call
+        self.user_defined_address = user_defined_address
+
+
+class StateChangeCallsAnnotation(StateAnnotation):
+    def __init__(self) -> None:
+        self.calls = []  # type: List[CallIssue]
+
+    def __copy__(self):
+        result = StateChangeCallsAnnotation()
+        result.calls = copy(self.calls)
+        return result
+
+
+class StateChange(DetectionModule):
+    """This module searches for state change after low level calls (e.g. call.value()) that
+    forward gas to the callee."""
+
+    def __init__(self):
+        """"""
+        super().__init__(
+            name="State Change After External calls",
+            swc_id=REENTRANCY,
+            description=DESCRIPTION,
+            entrypoint="callback",
+            pre_hooks=[
+                "CALL",
+                "SSTORE",
+                "DELEGATECALL",
+                "STATICCALL",
+                "CREATE",
+                "CALLCODE",
+            ],
+        )
+
+    def execute(self, state: GlobalState):
+        """
+
+        :param state:
+        :return:
+        """
+        self._issues.extend(self._analyze_state(state))
+        return self.issues
+
+    @staticmethod
+    def _add_external_call(
+        state: GlobalState, annotations: List[StateChangeCallsAnnotation]
+    ) -> None:
+        call = get_call_from_state(state)
+        gas = state.mstate.stack[-1]
+        to = state.mstate.stack[-2]
+        if call is None:
+            return
+        try:
+            constraints = copy(state.node.constraints)
+            solver.get_model(
+                constraints + [UGT(gas, symbol_factory.BitVecVal(2300, 256))]
+            )
+
+            # Check whether we can also set the callee address
+            try:
+                constraints += [to == 0xDEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEF]
+                solver.get_model(constraints)
+                annotations[0].calls.append(
+                    CallIssue(call=call, user_defined_address=True)
+                )
+            except UnsatError:
+                annotations[0].calls.append(
+                    CallIssue(call=call, user_defined_address=False)
+                )
+
+        except UnsatError:
+            pass
+
+    @staticmethod
+    def _analyze_state(state: GlobalState) -> List[Issue]:
+        """
+
+        :param state:
+        :return:
+        """
+        address = state.get_current_instruction()["address"]
+        annotations = cast(
+            List[StateChangeCallsAnnotation],
+            list(state.get_annotations(StateChangeCallsAnnotation)),
+        )
+        if len(annotations) == 0:
+            log.debug("Creating annotation for state")
+            state.annotate(StateChangeCallsAnnotation())
+            annotations = cast(
+                List[StateChangeCallsAnnotation],
+                list(state.get_annotations(StateChangeCallsAnnotation)),
```

Comment:
this annotation is also added for each sstore now

Reply:
sorry, i can't get it, can you elaborate?

Reply:
This function is executed for each callback, which includes callbacks for sstore insns. 
If there is a callback for a sstore insn no new annotation should be created